test/minikube/minikube_base_tde_test.go

```diff
@@ -0,0 +1,213 @@
+// +build enterprise
+
+package minikube
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/nuodb/nuodb-helm-charts/v3/test/testlib"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func applyStoragePasswordSecret(t *testing.T, namespaceName string, name string, passwords []string) {
+	assert.Greater(t, len(passwords), 0)
+	opts := k8s.NewKubectlOptions("", "", namespaceName)
+	var kubectlArgs []string
+	kubectlArgs = append(kubectlArgs, "create", "secret", "generic", name, "--dry-run=client", "-o", "yaml")
+	kubectlArgs = append(kubectlArgs, "--from-literal", fmt.Sprintf("target=%s", passwords[0]))
+	for i, password := range passwords[1:] {
+		kubectlArgs = append(kubectlArgs, "--from-literal", fmt.Sprintf("historical-%d=%s", i, password))
+	}
+	secret, err := k8s.RunKubectlAndGetOutputE(t, opts, kubectlArgs...)
+	require.NoError(t, err)
+	tmpfile, err := ioutil.TempFile("", "tde_secret")
+	require.NoError(t, err)
+	defer os.Remove(tmpfile.Name())
+	_, err = tmpfile.WriteString(secret)
+	require.NoError(t, err)
+	err = tmpfile.Close()
+	require.NoError(t, err)
+	k8s.RunKubectl(t, opts, "apply", "-f", tmpfile.Name())
+	testlib.AddTeardown(testlib.TEARDOWN_ADMIN, func() {
+		k8s.RunKubectlE(t, opts, "delete", "secret", name)
+	})
+}
+
+func TestAdminColdStartWithTDE(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	options := helm.Options{
+		SetValues: map[string]string{
+			"admin.tde.secrets.demo":                "demo-tde-secret",
+			"database.name":                         "demo",
+			"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"admin.tde.storagePasswordsDir":         "/etc/nuodb/encryption",
+		},
+	}
+	randomSuffix := strings.ToLower(random.UniqueId())
+	namespaceName := fmt.Sprintf("admin-cold-start-with-tde-%s", randomSuffix)
+	testlib.CreateNamespace(t, namespaceName)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	options.KubectlOptions = kubectlOptions
+
+	password := strings.ToLower(random.UniqueId())
+	applyStoragePasswordSecret(t, namespaceName, options.SetValues["admin.tde.secrets.demo"], []string{password})
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+	helmChartReleaseName, _ := testlib.StartAdmin(t, &options, 1, namespaceName)
```

Comment:
What happens if the secret is created after the Admin layer is started?

Reply:
The pods will stick in "Init" state with the following warning events:
``` 
Warning  FailedMount       5s (x5 over 13s)  kubelet            MountVolume.SetUp failed for volume "tde-volume-demo" : secret "demo-tde-secrett" not found
Warning  FailedMount       5s (x5 over 13s)  kubelet            MountVolume.SetUp failed for volume "tde-volume-demo2" : secret "demo2-tde-secrett" not found
```
They need to be restarted after needed secrets are created.